app/workers/scheduled_worker_recorder.rb

```diff
@@ -0,0 +1,7 @@
+module ScheduledWorkerRecorder
```

Comment:
The reason I think it makes sense to include this by default in the `ScheduledWorker` class and only record changes if `arg`, `kwargs` or `block` are passed is because the scheduled runner isn't set up to pass arguments to scheduled jobs.

And I think this is reasonable - scheduled jobs shouldn't receive arguments when they're run regularly, and the record functionality should only happen when they are run from the scheduler

Reply:
Following up on the offline discussion: 

We could use the following to conditionally record / not record based on the presence / absence of arguments:

```rb
[*args, *kwargs.values].all?(&:blank?)
```

But this won't do, because scheduled workers that do double-duty as enqueuers and enqueued need recording to happen when arguments are passed, but "single-duty" scheduled jobs need to record when no arguments are passed. 

Checking the arity of the `perform` also doesn't help — it's variable.

I'll ship as-is, prepending `ScheduledWorkerRecorder` only where it applies.

Reply:
> But this won't do, because scheduled workers that do double-duty as enqueuers and enqueued need recording to happen when arguments are passed, but "single-duty" scheduled jobs need to record when no arguments are passed

I don't believe this is accurate.

"Double duty" workers only should record when no arguments are passed. "Single duty" workers should only record when no arguments are passed.

Reply:
> "Double duty" workers only should record when no arguments are passed.

D'oh. Thanks—I think I got turned around somewhere down the rabbit hole. 

Updated in https://github.com/bikeindex/bike_index/pull/1299